src/NServiceBus.Hosting.Windows/GenericHost.cs

```diff
@@ -70,27 +71,67 @@ public void Start()
         }
 
         /// <summary>
-        ///     Finalize
+        /// Shut down bus or give up after 30 seconds, so host won't hang indefinitely.
         /// </summary>
         public void Stop()
         {
-            wcfManager.Shutdown();
+            Task disposeTask = DisposeBusTask();
 
-            if (bus != null)
-            {
-                bus.Dispose();
+            StartShutdownReporter();
+            var timeoutTask = Task.Delay(TimeSpan.FromSeconds(30));
```

Comment:
I guess we could ask the service manager for more time? 

http://stackoverflow.com/a/7441569/236004


Is it worth it?

How long would we keep on doing that 1 min 2min? more?